src/Tests/Tests/Test_Vibrato.cpp

```diff
@@ -0,0 +1,206 @@
+#include "MUSI6106Config.h"
+
+#ifdef WITH_TESTS
+#include "Vector.h"
+#include "Vibrato.h"
+
+#include "gtest/gtest.h"
+
+#include "RingBuffer.h"
+
+
+namespace vibrato_test {
+    void CHECK_ARRAY_CLOSE(float* buffer1, float* buffer2, int iLength, float fTolerance)
+    {
+        for (int i = 0; i < iLength; i++)
+        {
+            EXPECT_NEAR(buffer1[i], buffer2[i], fTolerance);
+        }
+    }
+
+    void RingBufferTest(int iRingBuffLength) {
```

Comment:
I don't know why these functions are not written in gtest framework like `Test_AudioIO.cpp`. Shouldn't it use `TEST_F` to define test functions? The setup should be done by inheriting `testing::Test` and override `SetUp()` method. The function here does not take any advantage of gtest framework, which does not fulfill the 0st requirement of the assignment.

> 0. [10] Implement the Ringbuffer as specified in exercise 3: ringbuffer with fractional delay but **move your tests to the gtest framework** (see below).

Reply:
And I don't know why ring buffer test was put in the namespace of `vibrato_test`?... I also looked around the code base and I don't see anywhere these functions are called and where parameters are fed. As long as these functions are not defined as `TEST_F`, I doubt if they will be found by [const UnitTest::TestList& allTests( UnitTest::Test::GetTestList() );](https://github.com/Jiarui-Xu-Gatech/2022-MUSI6106/blob/assignment2_vibrato/src/Tests/TestExec/Test.cpp#L90). What's more, **each test deletes INPUT FLOAT ARRAYS**. That is to say, after calling each function, the outside caller have to **reallocate** and **re-initialize** these arrays again, which does not make sense at all. It's also deleting something that is not allocated in the function body. The design here need full rewrite.